publisher/Rakefile

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+# rubocop:disable LineLength
+
+# Add your own tasks in files placed in lib/tasks ending in .rake,
+# for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
+
+require_relative 'config/application'
+require 'fileutils'
+
+Rails.application.load_tasks
+
+# Checks to see if any files already exist at the destination URI
+def guard_against_overwrite!(destination_uri)
+  sh "aws s3 ls #{destination_uri}" do |ok, res|
+    case res.exitstatus
+    when 0
+      # if the command is successful, we know files are present, which is an error condition for us
+
+      err_msg = <<~ERR_MSG
+        **** Pre-existing files were detected in '#{destination_uri}', aborting for safety reasons.
+        **** To overwrite the files, please invoke this task with the override argument:
+        **** rake publish[0.00.0, override]
+      ERR_MSG
+
+      abort err_msg
+      break
+    when 1
+      return true # exit status 1 means no files found, which is the success condition for this method
+    else
+      abort "Received unexpected error while checking for pre-existing files in '#{destination_uri}': #{res.exitstatus}.  Please try again."
+      break
+    end
+  end
+end
+
+desc 'Compile nfg_ui assets and upload to S3 for public access'
+task :publish, %i[nfg_ui_version override_mode] => %i[assets:precompile] do |_, nfg_ui_version: nil, override_mode: nil|
```

Comment:
Can you confirm the order of operations for `publish` to fire? Would we release first then publish?

Reply:
normally you would just call 'rake release' which would take care of invoking 'rake publish'. but if for some reason you wanted to publish as a separate step (like to correct an error) that could be done inside the publisher/ app using 'rake publish'

Reply:
I'd be in favor of having at least a basic test here that evaluates your error messages and statuses etc.

Reply:
Awesome! Love it.

Reply:
@subelsky would you update the Readme doc to include step by step instructions to publish in the event of an error as part of this PR? You can add a new section above "Usage"